EFCache.Redis/RedisCache.cs

```diff
@@ -167,75 +170,105 @@ public bool GetItem(string key, out object value)
 			}
 		}
 
+		// Note: This method does not use try/catch. It attempts retry and then throws if unsuccessful
+		// This method should never be allowed to proceed with an exception as it will lead to
+		// an incoherent cache
 		public void InvalidateSets(IEnumerable<string> entitySets)
 		{
 			// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
 			entitySets.GuardAgainstNull(nameof(entitySets));
 
 			var database = Connection.GetDatabase();
 
-			try
+			var keysAndEntitySetsToInvalidate = new HashSet<(string, RedisKey)>();
+			// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
+			foreach (var entitySet in entitySets)
 			{
-				var keysAndEntitySetsToInvalidate = new HashSet<(string, RedisKey)>();
-				// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
-				foreach (var entitySet in entitySets)
-				{
-					var entitySetKey = AddCacheQualifier(entitySet);
-					var keys = database.SetMembers(entitySetKey).Select(v => (v.ToString(), entitySetKey));
-					keysAndEntitySetsToInvalidate.UnionWith(keys);
-				}
-
-				var transaction = database.CreateTransaction();
-				foreach (var (key, entitySetKey) in keysAndEntitySetsToInvalidate)
-				{
-					var hashedKey = HashKey(key);
-					transaction.KeyDeleteAsync(hashedKey);
-					transaction.SetRemoveAsync(entitySetKey, hashedKey);
-				}
-				var committed = transaction.Execute();
+				var entitySetKey = AddCacheQualifier(entitySet);
+				var keys = database.SetMembers(entitySetKey).Select(v => (v.ToString(), entitySetKey));
```

Comment:
@jkdufair If none of these calls are wrapped in the try-catch anymore, what happens if one of them fails?

Reply:
Then the DB request should and will fail. Since, in our case, we also have retries on sk-net-js, it eventually recovers. I put a "chaos monkey" in for testing purposes and it performed well!

Reply:
![image](https://user-images.githubusercontent.com/623494/44870415-31f76400-ac5e-11e8-8d6e-b590937337da.png)